### ISSUE TITLE
`man`: Optional parts of arguments are not hightlighting

### ISSUE THREAD

**Issue by SuperSandro2000** [sharkdp/bat#1148]
**What version of `bat` are you using?**
bat 0.15.4

**Describe the bug you encountered:**
Optional parts of arguments in manpages are not highlighted.

I used the following commands:

```
export MANPAGER=sh -c 'col -bx | bat -l man -p'
git push --help
```

![image](https://user-images.githubusercontent.com/7258858/90928264-d8624400-e3f6-11ea-8e4c-d94aacfdde00.png)
![image](https://user-images.githubusercontent.com/7258858/90933897-9e963b00-e400-11ea-9384-ba4ee8de502b.png)

**Describe what you expected to happen?**
``--[no-]signed`` should be green and ``(true|false|if-asked)`` should be orange.

**How did you install `bat`?**
apt-get via GitHub release

**Comment by sharkdp** [comment]
Thank you for reporting this. The [`man` syntax definition](https://github.com/sharkdp/bat/blob/master/assets/syntaxes/02_Extra/Manpage.sublime-syntax) has been manually written for `bat` and is not extensively tested. Contributions are very welcome!

**Comment by SuperSandro2000** [comment]
I started gathering some examples and I don't think this is easily match able by regex:

```
--no-recurse-submodules, --recurse-submodules=check|on-demand|only|no
--[no-]thin
--repo=<repository>
--[no-]signed, --signed=(true|false|if-asked)
[--[no-]signed|--signed=(true|false|if-asked)]
[--all | --mirror | --tags] [--follow-tags] [--atomic] [-n | --dry-run] [--receive-pack=<git-receive-pack>]
```

Especially this line is hard to match:

``
[--[no-]signed|--signed=(true|false|if-asked)]
``

It translates to either arg A or B with optionals parts.

This is my attempt but I assume the regex is broken for other examples right now: ``(?:[^a-zA-Z0-9_-]|^|\s)(--?[A-Za-z0-9\[][A-Za-z0-9-\]]*)(?:(=)?("?)([A-Za-z0-9<>()|=-]+)("?))?``.

**Comment by sharkdp** [comment]
Thank you for looking into this. I'm sure we will not catch all cases, but we could try to solve some of the more often used ones.

The existing regex should probably be "cleaned up" first. Sublime syntax files allow for multiline regexes with comments.

**Comment by SuperSandro2000** [comment]
I think the main problem is that there is no clear and well defined syntax like in programming languages and it can be very program specific the highlighting. To do this properly we probably need more than just regexs.

**Comment by sharkdp** [comment]
Fix released in [bat v0.17](https://github.com/sharkdp/bat/releases/tag/v0.17.0).
